src/stores/room-list/filters/NameFilterCondition.ts

```diff
@@ -65,17 +65,7 @@ export class NameFilterCondition extends EventEmitter implements IFilterConditio
         return this.matches(room.name);
     }
 
-    private normalize(val: string): string {
-        // Note: we have to match the filter with the removeHiddenChars() room name because the
-        // function strips spaces and other characters (M becomes RN for example, in lowercase).
-        return removeHiddenChars(val.toLowerCase())
-            // Strip all punctuation
-            .replace(/[\\'!"#$%&()*+,\-./:;<=>?@[\]^_`{|}~\u2000-\u206f\u2e00-\u2e7f]/g, "")
-            // We also doubly convert to lowercase to work around oddities of the library.
-            .toLowerCase();
-    }
-
-    public matches(val: string): boolean {
-        return this.normalize(val).includes(this.normalize(this.search));
+    public matches(normalizedName: string): boolean {
```

Comment:
this gets called with `room.name` - that needs updating

Reply:
Its also called by RoomSubList - https://github.com/matrix-org/matrix-react-sdk/blob/54e139a6adc7604411f041912ec5b97fb4903dba/src/components/views/rooms/RoomSublist.tsx#L262 which also needs normalizing